pkg/api/latest/codec.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package latest
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/conversion"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/runtime/serializer/json"
+	"k8s.io/kubernetes/pkg/runtime/serializer/recognizer"
+)
+
+var (
+	Codecs                CodecFactory
+	UniversalDeserializer runtime.Decoder
+
+	jsonSerializer = json.NewSerializer(conversion.DefaultMetaFactory, api.Scheme, runtime.ObjectTyperToTyper(api.Scheme))
+	yamlSerializer = json.NewYAMLSerializer(conversion.DefaultMetaFactory, api.Scheme, runtime.ObjectTyperToTyper(api.Scheme))
+)
+
+type serializerType struct {
+	AcceptContentTypes []string
```

Comment:
This is not for end users, merely to enable local code to extend it (for
now).

On Mon, Nov 30, 2015 at 7:49 AM, David Eads <notifications@github.com>
wrote:

> In pkg/api/latest/codec.go
> <https://github.com/kubernetes/kubernetes/pull/17922#discussion_r46138851>
> :
>
> > +	"k8s.io/kubernetes/pkg/conversion"
> > +	"k8s.io/kubernetes/pkg/runtime"
> > +	"k8s.io/kubernetes/pkg/runtime/serializer/json"
> > +	"k8s.io/kubernetes/pkg/runtime/serializer/recognizer"
> > +)
> > +
> > +var (
> > +	Codecs                CodecFactory
> > +	UniversalDeserializer runtime.Decoder
> > +
> > +	jsonSerializer = json.NewSerializer(conversion.DefaultMetaFactory, api.Scheme, runtime.ObjectTyperToTyper(api.Scheme))
> > +	yamlSerializer = json.NewYAMLSerializer(conversion.DefaultMetaFactory, api.Scheme, runtime.ObjectTyperToTyper(api.Scheme))
> > +)
> > +
> > +type serializerType struct {
> > +	AcceptContentTypes []string
>
> Does this extend any interfaces? Seems like I'd want this to actually be
> an interface.
>
> Also, public fields on a private struct?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17922/files#r46138851>.
>